modules/porous_flow/src/kernels/PorousFlowDarcyBase.C

```diff
@@ -93,22 +100,53 @@ PorousFlowDarcyBase::computeQpResidual()
 void
 PorousFlowDarcyBase::computeResidual()
 {
+  computeProtoFlux();
+  if (_pf_count_used)
+  {
+    unsigned pf_count(_pf_count->maxUpDownSwaps(0));
+    Moose::out << "elem = " << _current_elem->id() << "pf count returns " << pf_count << "\n";
+  }
   upwind(JacRes::CALCULATE_RESIDUAL, 0.0);
 }
 
 void
 PorousFlowDarcyBase::computeJacobian()
 {
+  // Compute the fluxes the mobility terms because we need to know which
+  // nodes are upwind and which are downwind
+  if (!_pf_count_used)
+    computeProtoFlux();
   upwind(JacRes::CALCULATE_JACOBIAN, _var.number());
 }
 
 void
 PorousFlowDarcyBase::computeOffDiagJacobian(unsigned int jvar)
 {
+  // Compute the fluxes the mobility terms because we need to know which
+  // nodes are upwind and which are downwind
+  if (!_pf_count_used)
+    computeProtoFlux();
   upwind(JacRes::CALCULATE_JACOBIAN, jvar);
 }
 
 void
+PorousFlowDarcyBase::computeProtoFlux()
+{
+  const unsigned int num_nodes = _test.size();
+  for (unsigned ph = 0; ph < _num_phases; ++ph)
+  {
+    _proto_flux[ph].assign(num_nodes, 0);
+    for (_qp = 0; _qp < _qrule->n_points(); _qp++)
+    {
+      for (_i = 0; _i < num_nodes; ++_i)
+        _proto_flux[ph][_i] += _JxW[_qp] * _coord[_qp] * darcyQp(ph);
+    }
+  }
+  if (_pf_count_used)
+    _pf_count->setProtoFlux(_proto_flux);
```

Comment:
This seems really dangerous to me and is subject to breakage. In general in MOOSE you shouldn't be trying to shove data into other objects from some other object objects. With threading and decomposition, it's just a bad idea. In fact, our UserObject system, which is usually the extra storage dumping ground, simply doesn't allow this. I suspect you might be using ElementalAux to bypass that restriction.

Without really digging through all of this code, have you considered storing information right inside the kernel? Honestly that's not ideal either but is better then pushing information into another object in my opinion. Ideally UserObject-based solution would be my preferred method if possible, but I don't think that'll work particularly well since you are trying to track occurrences of something happening in a Kernel.